packages/extension/src/pages/setting/address-book/index.tsx

```diff
@@ -85,13 +85,21 @@ export const AddressBookPage: FunctionComponent<{
     const [dropdownOpen, setOpen] = useState(false);
     const toggle = () => setOpen(!dropdownOpen);
 
-    // TODO(!!!): This code needs reworking. Was not immediately clear what was
-    //            going on here and there was no typing information for me to
-    //            work with
     const [addAddressModalOpen, setAddAddressModalOpen] = useState(
       chatSectionParams.openModal || false
     );
     const [addAddressModalIndex, setAddAddressModalIndex] = useState(-1);
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
```

Comment:
Purpose of this useeffect?

Reply:
To set loader ( there is delay when the add address popup opens ) only if its coming from chat section to add address. to handle that we have added useEffect.also looking for better solution to handle this in meantime